File: Estructura_de_datos_Java/Recorrido_Columnas.java
public class Recorrido_Columnas {
    public static void main(String[]args){
        int [][] numeros = new int [3][3];
        //llenado del arreglo
        for (int i = 0; i < numeros.length; i++){
            for (int j = 0; j < numeros[0].length; j++){
                System.out.println("Ingrese un numero para la posicion " + i + "," + j);
                numeros[i][j] = Integer.parseInt(System.console().readLine());
            }
        }
        //recorrido por columnas
        //el ciclo externo recorre las columnas
        //el ciclo interno recorre las filas
        for (int j = 0; j < numeros[0].length; j++){
            for (int i = 0; i < numeros.length; i++){
                System.out.println(numeros[i][j]);
            }

        }
    }
}
    

File: Estructura_de_datos_Python/recorrido_Columnas.py
Matriz_numeros = [
    [1,2,3],
    [4,5,6],
    [7,8,9]
]
# Recorrer las columnas
for j in range(len(Matriz_numeros[1])):
    # Recorrer las filas
    for i in range(len(Matriz_numeros[0])):
        print(Matriz_numeros[i][j])